apps/src/applab/designElements/photoSelect.jsx

```diff
@@ -0,0 +1,187 @@
+          )}
+        />
+        <BooleanPropertyRow
+          desc={'hidden'}
+          initialValue={$(element).hasClass('design-mode-hidden')}
+          handleChange={this.props.handleChange.bind(this, 'hidden')}
+        />
+        <ZOrderRow
+          element={this.props.element}
+          onDepthChange={this.props.onDepthChange}
+        />
+      </div>
+    );
+  }
+}
+
+class PhotoChooserEvents extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onInsertEvent: PropTypes.func.isRequired
+  };
+
+  getPhotoSelectedEventCode() {
+    const id = elementUtils.getId(this.props.element);
+    const code = `onEvent("${id}", "change", function() {\n  console.log("${id} photo selected!");\n  console.log(getImageURL("${id}"));\n});\n`;
+    return code;
+  }
+
+  insertPhotoSelected = () =>
+    this.props.onInsertEvent(this.getPhotoSelectedEventCode());
+
+  render() {
+    const element = this.props.element;
+    const clickName = i18n.designElementPhotoSelectClickName();
+    const clickDescription = i18n.designElementPhotoSelectClickDescription();
+
+    return (
+      <div id="eventRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow={true}
+        />
+        <EventHeaderRow />
+        <EventRow
+          name={clickName}
+          desc={clickDescription}
+          handleInsert={this.insertPhotoSelected}
+        />
+      </div>
+    );
+  }
+}
+
+export default {
+  PropertyTab: PhotoChooserProperties,
+  EventTab: PhotoChooserEvents,
+  themeValues: themeValues.photoSelect,
+
+  create: function() {
+    const element = document.createElement('label');
+    element.setAttribute('class', 'img-upload fa fa-camera');
+    element.style.margin = '0';
+    element.style.borderStyle = 'solid';
+    element.style.overflow = 'hidden';
+
+    elementLibrary.setAllPropertiesToCurrentTheme(
+      element,
+      designMode.activeScreen()
+    );
+    element.style.padding = '0';
+    element.style.textAlign = 'center';
+    element.style.fontSize = '32px';
+
+    element.style.width = '75px';
+    element.style.height = '50px';
+    element.style.display = 'flex';
+    element.style.alignItems = 'center';
+    element.style.justifyContent = 'center';
+    const newInput = document.createElement('input');
+    newInput.type = 'file';
+    newInput.accept = 'image/*';
+    newInput.capture = 'camera';
+    newInput.hidden = true;
+    element.appendChild(newInput);
+    return element;
+  },
+  onDeserialize: function(element, updateProperty) {
+    // Disable image upload events unless running
+    $(element).on('click', () => {
+      if (Applab.isRunning()) {
+        element.childNodes[0].disabled = false;
+      } else {
+        element.childNodes[0].disabled = true;
+      }
```

Comment:
super nit: this conditional can be simplified to `element.childNodes[0].disabled = !Applab.isRunning();`